tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/workspace/NewLiferayWorkspaceWizard.java

```diff
@@ -0,0 +1,222 @@
+
+            final String message = wizardPage.getMessage();
+            final int messageType = wizardPage.getMessageType();
+
+            if( messageType == IMessageProvider.ERROR && !CoreUtil.isNullOrEmpty( message ) )
+            {
+                try
+                {
+                    if( LiferayWorkspaceUtil.hasLiferayWorkspace() )
+                    {
+                        wizardPage.setMessage(
+                            WorkspaceNameValidationService.hasLiferayWorkspaceMsg, SapphireWizardPage.ERROR );
+                    }
+                    else
+                    {
+                        wizardPage.setMessage( "Please enter the workspace name.", SapphireWizardPage.NONE );
+                    }
+                }
+                catch( CoreException e )
+                {
+                    wizardPage.setMessage( LiferayWorkspaceUtil.multiWorkspaceError, SapphireWizardPage.ERROR );
+                }
+
+                firstErrorMessageRemoved = true;
+            }
+        }
+
+        return wizardPages;
+    }
+
+    @Override
+    public void init( IWorkbench workbench, IStructuredSelection selection )
+    {
+    }
+
+    private void openLiferayPerspective( IProject newProject )
+    {
+        final IWorkbench workbench = PlatformUI.getWorkbench();
+
+        final IConfigurationElement element = new DelegateConfigurationElement( null )
+        {
+
+            @Override
+            public String getAttribute( String aName )
+            {
+                if( aName.equals( "finalPerspective" ) )
+                {
+                    return LiferayWorkspacePerspectiveFactory.ID;
+                }
+
+                return super.getAttribute( aName );
+            }
+        };
+
+        BasicNewProjectResourceWizard.updatePerspective( element );
+
+        // select and reveal
+        BasicNewResourceWizard.selectAndReveal( newProject, workbench.getActiveWorkbenchWindow() );
+    }
+
+    @Override
+    protected void performPostFinish()
+    {
+        super.performPostFinish();
+
+        final NewLiferayWorkspaceOp op = element().nearest( NewLiferayWorkspaceOp.class );
+
+        final IProject newProject = CoreUtil.getProject( op.getWorkspaceName().content() );
+
+        try
+        {
+            addToWorkingSets( newProject );
+        }
+        catch( Exception ex )
+        {
+            ProjectUI.logError( "Unable to add project to working set", ex );
+        }
+
+        openLiferayPerspective( newProject );
+
+        setProjectExplorerLayoutNestedEnabled();
+    }
+
+    private void setProjectExplorerLayoutNestedEnabled() {
+
+        try
+        {
+            final ICommandService commandService =
+                (ICommandService) PlatformUI.getWorkbench().getService( ICommandService.class );
+
+            Command command = commandService.getCommand( ProjectPresentationHandler.COMMAND_ID );
```

Comment:
once you have the command there is a way to get the handler class and execute it without having to call "new" on the handler class directly.  Eclipse command service should do that for you.